### PR TITLE
Add troubleshooting steps when endpoint type doesn't match model used

### PR DESCRIPTION
**Pull request by hario90**
### Motivation and Context

Please help reviewers and future users, providing the following information:
  1. Why is this change required? Helps developers that don't understand why they're getting a certain error from the LLM.
  2. What problem does it solve?
  3. What scenario does it contribute to?
  4. If it fixes an open issue, please link to the issue here.

### Description
An [issue](https://github.com/microsoft/semantic-kernel/issues/1564) I created while trying to run one of the starters with the gpt-35-turbo model received feedback from other developers that they ran into the same issue. Given that this is a common issue, I wanted to add troubleshooting steps closer to the starter code. Additionally, I made the CSharp Azure Function project endpoint type configurable such that it can use either a text or chat completion endpoint for developers who are limited to certain model types.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel-starters/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: sk-csharp-azure-functions/config/KernelSettings.cs [hunk 1]
```diff
@@ -7,6 +7,9 @@ internal class KernelSettings
 {
     public const string DefaultConfigFile = "config/appsettings.json";
 
+    [JsonPropertyName("endpointType")]
+    public string EndpointType { get; set; } = EndpointTypes.TextCompletion;
```
Comment (dmytrostruk) [inline sk-csharp-azure-functions/config/KernelSettings.cs:11]: If we make it as `ChatCompletion` by default, will it reduce error probability? If yes, then I think it worth to change it.

**Review by dmytrostruk — APPROVED** [review]

File: sk-csharp-azure-functions/config/KernelBuilderExtensions.cs [hunk 2]
```diff
@@ -13,11 +13,25 @@ internal static KernelBuilder WithCompletionService(this KernelBuilder kernelBui
         switch (kernelSettings.ServiceType.ToUpperInvariant())
         {
             case ServiceTypes.AzureOpenAI:
```
Comment (RogerBarreto) [inline sk-csharp-azure-functions/config/KernelBuilderExtensions.cs:15]: Suggest using when in this scenario:

```suggestion
            case ServiceTypes.AzureOpenAI when kernelSettings.EndpointType == EndpointTypes.TextCompletion:
```

Same for others

**Review by RogerBarreto — COMMENTED** [review]
Suggested changes

**Review by RogerBarreto — APPROVED** [review]
Very good. Thanks!

**Review by dmytrostruk — APPROVED** [review]

**Pull request merged**